vital.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <string>
+#include <set>
+#include "./vital.h"
+
+using std::cout;
+using std::wcout;
+using std::flush;
+using std::this_thread::sleep_for;
+using std::chrono::seconds;
+
+Vital::Vital(float vitalValue, const VitalBaseline& baseline) :
+    value(vitalValue), vitalbase(baseline) {
+}
+
+std::set<int>& Vital::collectInvalidCategories() const {
+    static auto numCategories{ static_cast<int>(vitalbase.levels.size()) };
+    static std::set<int> invalidCategoies{ 0, 1, numCategories - 1, numCategories };
+    return invalidCategoies;
+}
+
+int Vital::getCategory() const {
+    auto iter = std::lower_bound(vitalbase.levels.begin(), vitalbase.levels.end(), value);
+    if (value == vitalbase.low) {
```

Comment:
Amma, it took me 3 google searches to understand this, but it was worth it :)
But `value` is floating point, so a comparison with `==` will not always give you what you want. BTW, I didn't understand the significance of this `if` statement

Reply:
yes, I had also thought about the comparison of floats using ==, and thought about whether to implement more "careful comparision step" using a tolerance value + using round-Off to 2-decimal points for the individual category-bounds, but then left it; as we had done a similar discussion in one of your previous sessions regarding "comparison of doubles".

Reply:
Ok, you are right. Practically this may not have a very high impact... because the readings (from any IoT sensor) don't have that much precision as floating point. Quantization is always there.